### PR TITLE
Optimize/sanitization

### PR DESCRIPTION
**Pull request by KianMiridoozini**
Sanitization of all input.
Sanitization of image filename when uploading.
Generating unique filename for uploads

### CODE REVIEW

File: app/controllers/AuthController.php
Comment (jayb2302) [inline app/controllers/AuthController.php:1]: Don'nt we need trim for the email ? to remove accidental spaces, in the end for example?

File: app/controllers/CocktailController.php [hunk 1]
```diff
@@ -244,28 +244,40 @@ private function validateCocktailInput($data)
...
-            $allowedTypes = ['image/jpeg', 'image/png', 'image/webp'];
-            $fileType = mime_content_type($file['tmp_name']);
-            if (!in_array($fileType, $allowedTypes)) {
-                $errors[] = "Invalid image format. Allowed formats are JPEG, PNG, and WEBP.";
-            }
-            if (!move_uploaded_file($file['tmp_name'], $target_file)) {
-                $errors[] = "There was an error uploading the image.";
-            }
-            return $image;
+// Handle image upload for new cocktails
```
Comment (jayb2302) [inline app/controllers/CocktailController.php:247]: Can you explain what this file extension is  ?

File: app/controllers/CocktailController.php [hunk 2]
```diff
@@ -244,28 +244,40 @@ private function validateCocktailInput($data)
...
+
+        // Step 2: Check the file extension
+        $allowedTypes = ['jpeg', 'jpg', 'png', 'webp'];
+        if (!in_array($fileExtension, $allowedTypes)) {
+            $errors[] = "Invalid image format. Allowed formats are JPEG, PNG, and WEBP.";
+            return null; // Exit if file type is not allowed
+        }
+
+        // Step 3: Generate a unique filename to avoid conflicts
+        $image = bin2hex(random_bytes(8)) . '.' . $fileExtension;
```
Comment (jayb2302) [inline app/controllers/CocktailController.php:263]: What is this bin2hex ?

File: app/controllers/AuthController.php
Comment (KianMiridoozini) [reply app/controllers/AuthController.php:1]: if you are talking about the input, it goes through the controller which has the sanitize function on it.

the sanitize function inside helpers.php has been updated to implement trim as well.

so everything that goes through sanitize in controllers, get both htmlspecialchars and trimm

**Review by jayb2302 — APPROVED** [review]
I approve but lets think more about this image file names, if we should use hex or sanitized cocktailTitle and cocktailId for better seo.

File: app/controllers/CocktailController.php (hunk 2 above)
Comment (KianMiridoozini) [reply app/controllers/CocktailController.php:263]: it is to generate unique filenames for the uploaded images. it was implemented as a suggestion, though other methods may serve additional pros whilst maintaining security.

File: app/controllers/CocktailController.php (hunk 1 above)
Comment (KianMiridoozini) [reply app/controllers/CocktailController.php:247]: what file extension

**Pull request merged**